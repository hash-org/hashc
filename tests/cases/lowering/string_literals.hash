// stage=ir, args=-C dump=ir -C ir-dump-mode=pretty


main := () => {
    print("Hello world");
    print("Hola mundo");
    print("Привет мир");
    print("你好，世界");
    print("Hallo Welt");
    print("Bonjour le monde");
}
