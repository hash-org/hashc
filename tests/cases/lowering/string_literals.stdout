
IR dump for function `main` defined at $DIR/string_literals.hash:4:9-11:2
main := () -> () {
    mut _0: ();
    _1: ();
    _2: ();
    _3: ();
    _4: ();
    _5: ();
    _6: ();

    bb0 {
        _1 = print(const "Hello world") -> bb1;
    }

    bb1 {
        _2 = print(const "Hola mundo") -> bb2;
    }

    bb2 {
        _3 = print(const "Привет мир") -> bb3;
    }

    bb3 {
        _4 = print(const "你好，世界") -> bb4;
    }

    bb4 {
        _5 = print(const "Hallo Welt") -> bb5;
    }

    bb5 {
        _6 = print(const "Bonjour le monde") -> bb6;
    }

    bb6 {
        _0 = ();
        return;
    }
}

